cleanlab/classification.py

```diff
@@ -263,7 +263,7 @@ def __init__(
     def fit(
         self,
         X,
-        labels,
+        y=None,
```

Comment:
@elisno After discussion with @cgnorthcutt we would like to do the following new plan. Sorry for the extra work and change of plans. Please ignore the old comments that contradict this new plan!

Reply:
**New plan**:  Keep the `labels` arg as is. And keep `labels` throughout the documentation.  This keeps the CleanLearning code args the same as for the rest of the package. Especially for `CleanLearning.find_label_issues()` we prefer the argname `labels` since that matches the argname for `filter.find_label_issues()`.  So this basically corresponds to reverting all the changes and going back to the what the code looked like before. 

That said, we want this code to work with arbitrary sklearn code the user may be using, such that the user can drop in CleanLearning in place of the sklearn model without changing anything.  This means that at least `CleanLearning.fit()` must be able support the arg `y` in-case the user's sklearn code was doing something like: `SklearnEstimator.fit(X=...,y=...)`.

What this means is the only change we want this PR to add from what the code looked like before is support for an extra arg: `y`.

This extra arg should appear at the end of the args list of `CleanLearning.fit()` and it should have docstring that looks something like this: 
```
y: np.array or pd.Series, optional
  Alternative argument that can be specified instead of `labels`. Specifying `y` has the same effect as specifying `labels`, and is offered as an alternative for compatibility with sklearn. 
```
New plan:  Keep the `labels` arg as is. And keep `labels` throughout the documentation.  This keeps the CleanLearning code args the same as for the rest of the package. Especially for `CleanLearning.find_label_issues()` we prefer the argname `labels` since that matches the argname for `filter.find_label_issues()`.  So this basically corresponds to reverting all the changes and going back to the what the code looked like before. 

That said, we want this code to work with arbitrary sklearn code the user may be using, such that the user can drop in CleanLearning in place of the sklearn model without changing anything.  This means that at least `CleanLearning.fit()` must be able support the arg `y` in-case the user's sklearn code was doing something like: `SklearnEstimator.fit(X=...,y=...)`.

What this means is the only change we want this PR to add from what the code looked like before is support for an extra arg: `y`.

This extra arg should appear at the end of the args list of `CleanLearning.fit()` and it should have docstring that looks something like this: 
```
y: np.array or pd.Series, optional
  Alternative argument that can be specified instead of `labels`. Specifying `y` has the same effect as specifying `labels`, and is offered as an alternative for compatibility with sklearn. 
```

Your code should still raise ValueError if `y` and `labels` are both specified, stating only one should be specified.

All other functions in this file besides `fit()` should only take in arg `labels`, not `y`. So fit() should just internally convert `labels = y` as soon as it sees `y` was specified.

Reply:
Also, I'd suggest you make `y` a [keyword-only argument](https://peps.python.org/pep-3102/).

Reply:
Note that for this to work, `labels` would have to be "optional" to allow the use of `CleanLearning.fit(X=..., y=...)`

Reply:
Yep exactly, we do need to change `labels` to be optional arg. Also should raise ValueError if `labels` and `y` are both None.

Reply:
One thing I have yet to look into: How about writing a decorator for aliasing `labels`? That way it could be explicitly "required" in the signature.

Reply:
Should we make this change throughout tutorials ex. image and tabular tutorials call  `find_label_issues(labels=y, ...)`

Reply:
@ulya-tkch i think its fine to still use `labels=labels`. but if your tutorial ever uses sklearn APIs on a CleanLearning object, then we must use `y=labels` to avoid an error/for compatibilty.